homeassistant/components/soundtouch/__init__.py

```diff
@@ -1 +1,142 @@
 """The soundtouch component."""
+import logging
+
+from libsoundtouch import soundtouch_device
+from libsoundtouch.device import SoundTouchDevice
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, Platform
+from homeassistant.core import HomeAssistant, ServiceCall
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.typing import ConfigType
+
+from .const import (
+    DOMAIN,
+    SERVICE_ADD_ZONE_SLAVE,
+    SERVICE_CREATE_ZONE,
+    SERVICE_PLAY_EVERYWHERE,
+    SERVICE_REMOVE_ZONE_SLAVE,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+SERVICE_PLAY_EVERYWHERE_SCHEMA = vol.Schema({vol.Required("master"): cv.entity_id})
+SERVICE_CREATE_ZONE_SCHEMA = vol.Schema(
+    {
+        vol.Required("master"): cv.entity_id,
+        vol.Required("slaves"): cv.entity_ids,
+    }
+)
+SERVICE_ADD_ZONE_SCHEMA = vol.Schema(
```

Comment:
Media player has built-in support for join and unjoin. we should use that.

Reply:
I just ported this over from the previous implementation because I didn't want to get this PR even more complicated.
Join and Unjoin services are on my to-do list for a follow-up PR.